node_modules/oae-principals/lib/util.js

```diff
@@ -190,7 +190,8 @@ var hideUserData = module.exports.hideUserData = function(ctx, user) {
     }
 
     // Hide the sensitive profile information if the user has limited access
-    if ((user.visibility === AuthzConstants.visibility.LOGGEDIN && !isLoggedIn) ||
+    if ((user.deleted ||
+        user.visibility === AuthzConstants.visibility.LOGGEDIN && !isLoggedIn) ||
         (user.visibility === AuthzConstants.visibility.PRIVATE && !isTargetUser)) {
```

Comment:
Can you please wrap some of these booleans into `isSomething` variables to make the if statement a bit more understandable?